src/OrchardCore.Modules/OrchardCore.Settings/Services/TenantEventHandler.cs

```diff
@@ -0,0 +1,29 @@
+using OrchardCore.Environment.Shell;
+using OrchardCore.Modules;
+
+namespace OrchardCore.Settings.Services;
+
+internal sealed class TenantEventHandler : ModularTenantEvents
```

Comment:
I think this is the only think that is needed here to fix the issue. The rest of the changes should not be needed.